src/LinuxCaptureService/ExtractSignalFromMinidump.cpp

```diff
@@ -0,0 +1,103 @@
+
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ExtractSignalFromMinidump.h"
+
+#include <string>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/Result.h"
+
+namespace orbit_linux_capture_service {
+
+namespace {
+
+// The constants and structs below help to interpret the minidump file format. The data here is
+// limited to access the part of the file that we are interested in. For a complete parser see
+// minidump-2-core.cc from https://chromium.googlesource.com/breakpad/breakpad/.
+
+uint32_t kStreamCountOffset = 8;
+uint32_t kStreamDirectoryOffset = 12;
+uint32_t kMdExceptionStream = 6;
+
+struct MDLocationDescriptor {
+  uint32_t data_size;
+  uint32_t rva;
+};
+
+struct MDRawDirectory {
+  uint32_t stream_type;
+  MDLocationDescriptor location;
+};
+
+struct MDException {
+  uint32_t exception_code;
+  uint32_t exception_flags;
+  uint64_t exception_record;
+  uint64_t exception_address;
+  uint32_t number_parameters;
+  uint32_t __align;
+  uint64_t exception_information[15];
+};
+
+struct MDRawExceptionStream {
+  uint32_t thread_id;
+  uint32_t __align;
+  MDException exception_record;
+  MDLocationDescriptor thread_context;
+};
+
+template <typename T>
+ErrorMessageOr<const T*> DataOrError(const std::string& content, uint64_t offset) {
+  if (content.size() < offset + sizeof(T)) {
+    return ErrorMessage("Unexpected end of data.");
+  }
+  return reinterpret_cast<const T*>(content.data() + offset);
+}
+
+template <typename T>
+ErrorMessageOr<const T*> ArrayElementOrError(const std::string& content, uint64_t offset,
```

Comment:
No strong feeling, just a suggestion: what about prepending these functions' names with `GetPointerTo`?

Reply:
Will keep as is merely for brevity.